### PR TITLE
fix: set preferences after re-import wallet

### LINKED ISSUE #231: Avoid set preferences error when db is deleted
Refactor get wallet from db and set preferences of current wallet and current address.

### PR DESCRIPTION
**Pull request by parodyBit**
Closes #231

### CODE REVIEW

**Comment by Tommytrg** [comment]
@parodyBit Could you run `dart format .`?

File: lib/util/preferences.dart [hunk 1]
```diff
@@ -27,10 +27,11 @@ class ApiPreferences {
     walletStorage.setCurrentWallet(walletId);
     String address;
     int index = int.parse(addressIndex.split("/").last);
-    if (addressIndex.split('/').first == "0") {
+    if (addressIndex.split('/').first == "0" &&
```
Comment (Tommytrg) [inline lib/util/preferences.dart:30]: Could you add a comment explaining what we are checking in this `if` condition?

**Review by gabaldon — CHANGES_REQUESTED** [review]

File: lib/bloc/crypto/crypto_bloc.dart [hunk 2]
```diff
@@ -159,10 +159,10 @@ class CryptoBloc extends Bloc<CryptoEvent, CryptoState> {
       internalIndex += 1;
     }
     ApiDatabase database = Locator.instance.get<ApiDatabase>();
+    await database.loadWalletsDatabase();
     await ApiPreferences.setCurrentWallet(_wallet.id);
```
[the remark this answers is not in the document]
Comment (Tommytrg) [reply lib/bloc/crypto/crypto_bloc.dart:163]: I see this error in `main` branch but it's solved in this issue. How could we reproduce it?

[the remark this answers is not in the document]
Comment (gabaldon) [reply lib/bloc/crypto/crypto_bloc.dart:163]: My bad, the error is solved in this branch

**Review by gabaldon — APPROVED** [review]

**Pull request merged**